nssp_demo/collate_score_tables.R

```diff
@@ -0,0 +1,202 @@
+script_packages <- c(
+  "data.table",
+  "argparser",
+  "stringr"
+)
+
+## load in packages without messages
+purrr::walk(script_packages, \(pkg) {
+  suppressPackageStartupMessages(
+    library(pkg, character.only = TRUE)
+  )
+})
+
+
+#' Get all the subdirectories within a parent directory
+#' that match the pattern for a forecast run for a
+#' given disease and optionally a given report date.
+#'
+#' @param parent_dir Directory in which to look for forecast subdirectories.
+#' @param diseases Names of the diseases to match, as a vector of strings,
+#' or a single disease as a string.
+#' @return A vector of paths to the forecast subdirectories.
+get_all_forecast_dirs <- function(dir_of_forecast_date_dirs,
+                                  diseases) {
+  # disease names are lowercase by convention
+  match_patterns <- str_c(tolower(diseases), "_r", collapse = "|")
+
+  dirs <- tibble::tibble(
+    dir_path = fs::dir_ls(
+      dir_of_forecast_date_dirs,
+      type = "directory"
+    )
+  ) |>
+    dplyr::filter(str_starts(
+      fs::path_file(dir_path),
+      match_patterns
+    )) |>
+    dplyr::pull(dir_path)
+
+  return(dirs)
+}
+
+
+process_loc_date_score_table <- function(model_run_dir) {
+  table_path <- fs::path(model_run_dir,
+    "score_table",
+    ext = "rds"
+  )
+  parsed <- hewr::parse_model_run_dir(model_run_dir)
+
+  if (!(fs::file_exists(table_path))) {
+    warning(glue::glue(
+      "No `score_table.rds` found for location ",
+      "{location} in directory {model_run_dir}"
+    ))
+    return(NULL)
+  }
+  score_table <- readr::read_rds(table_path)
+
+  for (x in names(parsed)) {
+    score_table$quantile_scores[[x]] <- parsed[[x]]
+    score_table$sample_scores[[x]] <- parsed[[x]]
```

Comment:
is this right? Why are they assigned the same thing?

Reply:
this is just adding metadata parsed from the path

Reply:
I can add a comment.

Reply:
Done in d9f0819 though it strikes me that this annotation probably belongs in `score_forecast.R`. Can move it in a separate PR